transforms/ray_cyl_intersection.py

```diff
@@ -28,24 +29,96 @@ def unit_vector(invec):
 
 def ray_plane(rays, rmat, tvec):
     """
+    tvec: Is the normal defining the plane.
+    rvec
     """
     nhat = rmat[:, 2].reshape(3, 1)
 
     rays = np.atleast_2d(rays)  # shape (npts, 3)
     npts = len(rays)
 
-    output = np.nan*np.ones_like(rays)
-
     numerator = np.tile(np.dot(tvec, nhat), npts)
     denominator = np.dot(rays, nhat).flatten()
-
     can_intersect = denominator < 0
-    u = np.tile(numerator[can_intersect]/denominator[can_intersect], (3, 1))
+
+    u = np.tile(numerator[can_intersect] / denominator[can_intersect], (3, 1))
+
+    output = np.nan*np.ones_like(rays)
     output[can_intersect, :] = rays[can_intersect] * u.T
 
     return np.dot(output - tvec, rmat)[:, :2]
 
 
+def tf_ray_plane(rays, rmat, tvec):
+    """
+    In general, the plane equation is defined by
+    $$\hat{n} \cdot (\vec{x} - \vec{x}_0)  = 0$$
+
+    where $\vec{x}_0$ defines the origin of rotation for the plane, $\hat{n}$
+    defines the plane normal.
+
+    For a given ray, $\vec{r}(t) = \vec{p}_0 + \hat{u} t$
+
+    the ray-plane intersection is defined by
+
+    $$t = \frac{\hat{n} \cdot (\vec{x}_0 - \vec{p}_0)} {\hat{n} \cdot \hat{u}} $$
+
+
+    To parameterize the detector plane, we will give the origin point a
+    translation vector.
+
+    This gives us
+
+    $$ \mathbf{R} \hat{n} \cdot (\vec{x} - (\vec{x}_0 + \vec{d}))  = 0$$
+
+    where $\vec{d}$ is the translation vector and $\mathbf{R}$ is the rotation
+    of the detector about its origin.
+
+
+    For this function, we'll take \vec{x_0} to be at the origin. We will define
+    the translation and rotation as `tvec` and `rmat`
+
+
+    :param tvec: Translation vector for the origin of the plane.
+    :param rmat: Rotation of the detector plane about its origin.
+    :param rays: Ray vector with $\vec{p}_0$ taken to be coordinate origin.
+
+    :return: The list of x, y intersection location of the plane.
+
+    As a convention, for tvec = [0, 0, 0], and R = np.eye(3), the detector
+    plane is located at the origin, with its normal pointing in the [0, 0, 1]
+    direction.
+
+    Similarly, by convention, the origin of the ray, $\vec{p}_0$ is taken to
+    be the origin as well.
+
+    """
+
+    # Detector normal, rotated about its origin.
+    nhat = tf.convert_to_tensor(rmat[:, 2], dtype='float32')
```

Comment:
Gotta remove this conversions and replace them with static checks. Input into this function needs to be `tf.tensors`